### PR TITLE
Fix #13883 | SelectBox.js with grouping (optgroup elements)

### PR DESCRIPTION
**Pull request by jigneshkotadiya**
- current implementation ignores optgroup elements in selects.
- Here is an update of SelectBox.js which takes optgroups in selects.

### CODE REVIEW

File: django/contrib/admin/static/admin/js/SelectBox.js [hunk 1]
```diff
@@ -1,136 +1,126 @@
...
-            for (var i = 0; i < boxOptionsLength; i++) {
-                boxOptions[i].selected = 'selected';
-            }
+            $("#" + id + " option").attr('selected', 'selected');
         }
-    };
+    }
     window.SelectBox = SelectBox;
-})();
+})(django.jQuery);
```
Comment (pope1ni) [inline django/contrib/admin/static/admin/js/SelectBox.js:126]: The use of jQuery is slowly being phased out within Django. Please find a way to avoid adding it back in.

File: django/contrib/admin/static/admin/js/SelectBox.js [hunk 2]
```diff
@@ -1,136 +1,126 @@
...
+            }
         },
         select_all: function(id) {
-            var box = document.getElementById(id);
-            var boxOptions = box.options;
-            var boxOptionsLength = boxOptions.length;
-            for (var i = 0; i < boxOptionsLength; i++) {
-                boxOptions[i].selected = 'selected';
-            }
+            $("#" + id + " option").attr('selected', 'selected');
```
Comment (pope1ni) [inline django/contrib/admin/static/admin/js/SelectBox.js]: This change is unrelated. It is restoring usage of jQuery which is undesirable.

File: django/contrib/admin/static/admin/js/SelectBox.js (hunk 1 above)
Comment (jigneshkotadiya) [reply django/contrib/admin/static/admin/js/SelectBox.js:126]: I will work on that

File: django/contrib/admin/static/admin/js/SelectBox.js (hunk 2 above)
Comment (jigneshkotadiya) [reply django/contrib/admin/static/admin/js/SelectBox.js]: restored

**Comment by jigneshkotadiya** [comment]
- I was removed jquery form SelectBox.js, and it's work fine with a grouped and ungrouped option, and js code checked with https://jshint.com/  
- still, pull-requests-javascript checks fail, 
can someone explain what needs to fix

**Comment by pope1ni** [comment]
We use `eslint`, not `jshint`. See [here](https://djangoci.com/job/pull-requests-javascript/5398/console) for failures.

In addition, you'll likely need to update the JavaScript unit tests under the `js_tests/admin/` directory.

**Comment by jigneshkotadiya** [comment]
> We use `eslint`, not `jshint`. See [here](https://djangoci.com/job/pull-requests-javascript/5398/console) for failures.
> 
> In addition, you'll likely need to update the JavaScript unit tests under the `js_tests/admin/` directory.

- are we need update` jshint` with `eslint` in this documentation https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/javascript/
- and I fixed lint error and test case error is there any other update

File: js_tests/admin/SelectBox.test.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ QUnit.test('init: no options', function(assert) {
     var $ = django.jQuery;
     $('<select id="id"></select>').appendTo('#qunit-fixture');
     SelectBox.init('id');
-    assert.equal(SelectBox.cache.id.length, 0);
+    assert.equal(SelectBox.cache.id.length, 1);
```
Comment (pope1ni) [inline js_tests/admin/SelectBox.test.js:11]: At a glance I'm unsure why this should change...

Comment (jigneshkotadiya) [reply js_tests/admin/SelectBox.test.js:11]: when console SelectBox.cache.id.length i got 1 so , I'll update with 1,
anyway I'll fix that 
thanks

**Comment by jigneshkotadiya** [comment]
> I think that you'll need to write new unit tests to cover the case when option groups are being used. I'm not convinced their should be any changes to existing tests.

- I was revert changes in an existing test case and add a new test case named  `option with group`
- in case if new addition to test-case and code modification try to give me detailed information for an update.

Thanks for guiding @pope1ni
